apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts

```diff
@@ -213,9 +215,36 @@ export class PublicFlightController {
       )
     }
 
-    const meetsADSRequirements = this.flightService.isADSPostalCode(
+    let meetsADSRequirements = this.flightService.isADSPostalCode(
       user.postalcode,
     )
+
+    // TODO: this is a quickly made temporary hotfix and should be rewritten
+    // along when the nationalregistry module is rewritten for the client V2 completely
+    if (!meetsADSRequirements) {
```

Comment:
You might consider adding a age check as well incase the edge case where user is actually adult and shouldn't be able to proceed.

Reply:
The age check already takes place for when the cache is set here:
https://github.com/island-is/island.is/blob/d50c699ab9237f809a8c8aa643f110c47e1d080e/apps/air-discount-scheme/backend/src/app/modules/user/user.service.ts#L95-L111

Reply:
Up to you.
I reckon its cheaper to check age vs search cache.
Every user that has !adsRequirements will search cache.

Reply:
Ah, that however is a good point

Reply:
Fixed